### ISSUE TITLE
Providing option to exit the App with double back presses (was: in the drop down menu).

### ISSUE THREAD

**Issue by Rahul5758** [kiwix/kiwix-android#678]
1. Provide an exit button in the drop down menu to exit the Kiwix app from any screen.
2. The user may press the back button accidentally while on the home screen so exit the app only after double back press.

Provide a exit option in the drop down so that the user may exit the app from any tab.

![exit_button_eg](https://user-images.githubusercontent.com/26683036/38021986-edefe8cc-329b-11e8-9564-2cf74e19e167.png)

Something like shown above.

Right now when the back button is pressed once while on the home screen the application closes on single back press. This may also happen accidentally so on single back press a toast should be displayed with a message similar to  "Please click Back again to Exit."
![exit_example](https://user-images.githubusercontent.com/26683036/38020784-bf562cea-3298-11e8-8c94-11e07563a1e3.PNG)

**Comment by brijeshshah13** [comment]
@Rahul5758 I think adding the facility to exit the app on double back presses would be an improvement. However, I don't think there is any need of adding the Exit menu option as the user can always close the app directly via the navigation bar of the device, can't he?

**Comment by kelson42** [comment]
We won't do that. "Exiting" an app has almost no sense on Android.

**Comment by Rahul5758** [comment]
@kelson42 You are right, but as @brijeshshah13 agreed can the facility to allow double press before actually exiting or minimizing the app be given?

**Comment by kelson42** [comment]
@Rahul5758 Ooops, sorry for this bad moove. I have reopen the ticket and changed the title.

**Comment by brijeshshah13** [comment]
@Rahul5758 Please go ahead and send a PR for the same. I'm assigning the issue to you.

**Comment by abdulwd** [comment]
Pressing back twice to exit fullscreen mode makes more sense. But pressing back twice every time to exit the app will be annoying to the user.

**Comment by mhutti1** [comment]
Yeah I don't think locking the user into the app is a good idea.

**Comment by Rahul5758** [comment]
What is your suggestion then @mhutti1?

**Comment by brijeshshah13** [comment]
@mhutti1 @abdulwd @Rahul5758 I think we must provide this facility only when/after the user is on the Home page of the ZIM file, this way the user won't have to back press twice every now and then.

**Comment by Rahul5758** [comment]
@brijeshshah13 I have stated this thing in the issue description.

**Comment by brijeshshah13** [comment]
@Rahul5758 You didn't clearly mention which Home page were you speaking of and you attached the Help page to depict it. I'm speaking of the Home page of the ZIM file.

**Comment by Rahul5758** [comment]
@mhutti1  SHAREit, Flipkart, Paytm are some apps that use this.

**Comment by mhutti1** [comment]
I don't have any experience with any of those. Most apps I use e.g YouTube, Gmail...

**Comment by srv-twry** [comment]
@mhutti1 @Rahul5758 @brijeshshah13 @kelson42 
I don't think it is a standard to add this double press to exit. It breaks the user experience. None of the official google apps implement them. Flipkart, Paytm do implement them but they are wrong here. 
One should only prompt the user, if there is something going on in the background and which will be closed if the app is closed. 
If i am a user and i accidently close kiwix, why can't i just reopen it again ?
Here are a few articles with much more insight - [first](https://www.androiddesignpatterns.com/2012/08/exit-application-dialogs-are-evil-dont.html)
[second](https://plus.google.com/+CyrilMottier/posts/EiXqUDrr6jT)

**Comment by Rahul5758** [comment]
@srv-twry The links you have given pertain to some different feature than the one I have suggested. There is no dialog box in my suggestion that annoys the user, just a toast message that press again to exit. This double press is suggested not to lock the user in the app but to prevent accidental exit from the app while on the home screen. If we are on some other screen accidental back press only leads to the previous fragment but on the home screen accidental back press would minimize the app, so this feature minimize on accidental back press.

**Comment by srv-twry** [comment]
@Rahul5758 Even though my links refer to the dialog only, the whole idea of confirm to exit is wrong. 

> This double press is suggested not to lock the user in the app but to prevent accidental exit from the app while on the home screen.

I understood it but why would you want to prevent accidental exit from the app ? Is there any loss for the user ? If he/she accidently pressed the back button, they can easily **reopen** the app. 
Let's say you accidently press the back button 1 out of 100 times you use the app. The remaining 99 times you press the back button when you actually want to exit the app. So your feature would mean an extra **annoying** step for the 99 times when you actually wanted to exit the app.

**Comment by harshvardhan-takawale** [comment]
I have an opinion here, let me know if you all agree. So when we open the app for the first time, it opens in the help activity and then we a guided to open a zim. Here in the summary page when you press back the app exits. So for a new user it is somewhat confusing as the point of entry and that of exit is different. So we can use this double back feature for this case.

**Comment by srv-twry** [comment]
@harshvardhan-takawale I agree that this is a bit confusing but adding the double back press feature is a workaround, not a solution. 
Solution would be to provide proper back stack navigation. Infact there is a GSOC idea on improving the overall experience of the user, hence if it worked upon in the summer, it can be tackled then.

**Comment by mhutti1** [comment]
@harshvardhan-takawale I think the solution to your issue would be having the help page be at the bottom of the stack for summaries but I don't think its that major of an issue. Closing this issue as won't implement.
